### DISCUSSION TITLE
[gatsby-source-graphql] "Error: Source GraphQL API: HTTP error 400 Bad Request" when trying to deploy...

### DISCUSSION THREAD

**Discussion by mikeyfe6** [gatsbyjs/gatsby#34697] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

**Trying to deploy my website to Netlify (from dev to prod), but it hangs when trying to connect to Strapi v4.0.6**

### Reproduction Link

https://github.com/mikeyfe6/Afrodiasphere

### Steps to Reproduce

1. Trying to deploy to Netlify through Github repo
2. My build fails in Netlify
3.
...

### Expected Result

Going through with the production process and deploy my website on Netlify

### Actual Result

I'm getting this errormessage: "Error: Source GraphQL API: HTTP error 400 Bad Request" when trying to deploy... 

my gatsby-config.js:

```
{
      resolve: "gatsby-source-graphql",
      options: {
        // Arbitrary name for the remote schema Query type
        typeName: "INSTANTIE",
        // Field under which the remote schema will be accessible. You'll use this in your Gatsby query
        fieldName: "instantie",
        // Url to query from
        url: `${process.env.GATSBY_BASE_URL}/graphql`,
      },
    },
```

<img width="1447" alt="Schermafbeelding 2022-02-02 om 20 22 19" src="https://user-images.githubusercontent.com/30863830/152222968-95e6b61e-e9ad-4fa3-99ad-86486b182304.png">

Also it complains about not finding certain modules......

<img width="1261" alt="Schermafbeelding 2022-02-02 om 20 22 40" src="https://user-images.githubusercontent.com/30863830/152222997-35b5325b-c942-4259-bd85-317c2a73fd37.png">

### Environment

```shell
System:
    OS: macOS 12.2
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 97.0.4692.99
    Safari: 15.3
  npmPackages:
    gatsby: ^4.6.1 => 4.6.1
    gatsby-image: ^3.11.0 => 3.11.0
    gatsby-plugin-catch-links: ^4.6.0 => 4.6.0
    gatsby-plugin-create-client-paths: ^4.6.0 => 4.6.0
    gatsby-plugin-manifest: ^4.6.0 => 4.6.0
    gatsby-plugin-netlify: ^4.1.0 => 4.1.0
    gatsby-plugin-nprogress: ^4.6.0 => 4.6.0
    gatsby-plugin-offline: ^5.6.0 => 5.6.0
    gatsby-plugin-react-helmet: ^5.6.0 => 5.6.0
    gatsby-plugin-sass: ^5.6.0 => 5.6.0
    gatsby-plugin-sharp: ^4.6.0 => 4.6.0
    gatsby-plugin-sitemap: ^5.6.0 => 5.6.0
    gatsby-plugin-styled-components: ^5.6.0 => 5.6.0
    gatsby-source-filesystem: ^4.6.0 => 4.6.0
    gatsby-source-graphql: ^4.6.0 => 4.6.0
    gatsby-source-strapi: ^1.0.2 => 1.0.2
    gatsby-transformer-sharp: ^4.6.0 => 4.6.0
```

### Config Flags

I used to, but didn't think i would need them anymore.... maybe thats's the problem??

**Comment by LekoArts** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Your URL might be incorrect. Does it work locally?

**Reply by mikeyfe6** [reply]
Hi @LekoArts, 

**I double-checked my URLs but couldn't find anything wrong.. it connects tho for some reason, because the live back-end gets a hit from my live frontend. It just doesn't want to proceed the production process, why.. I don't know.... See picture below:**

<img width="1199" alt="Schermafbeelding 2022-02-03 om 16 15 59" src="https://user-images.githubusercontent.com/30863830/152371220-3fe84082-112c-459d-9bff-fe3e7cb0b7aa.png">


**Locally it works only when I use `npm run develop`. It refuses to also `npm run build` my project.. showing this error, so its not that crazy that on Netlify it would do the same thing:**

<img width="1055" alt="Schermafbeelding 2022-02-03 om 16 11 17" src="https://user-images.githubusercontent.com/30863830/152370807-53e6e62a-6da9-4cb7-9344-027729b5be8c.png">

**Reply by LekoArts** [reply]
Try using Node 16 locally (you're using Node 17). If it works in `develop` you might be missing env vars for `gatsby build`
